src/main/java/co/rsk/federate/FederationProviderFromFederatorSupport.java

```diff
@@ -93,7 +97,7 @@ public Optional<Federation> getRetiringFederation() {
         Integer federationSize = federatorSupport.getRetiringFederationSize();
         Optional<Address> optionalRetiringFederationAddress = getRetiringFederationAddress();
 
-        if (federationSize == -1 || !optionalRetiringFederationAddress.isPresent()) {
+        if (federationSize == FEDERATION_NON_EXISTENT.getCode() || !optionalRetiringFederationAddress.isPresent()) {
```

Comment:
```suggestion
        if (federationSize == FEDERATION_NON_EXISTENT.getCode() || optionalRetiringFederationAddress.isEmpty()) {
```

Please check IDE warnings and suggestions

Reply:
Thanks, updated